src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -769,35 +742,43 @@ async Task ITokenCacheInternal.RemoveAccountAsync(IAccount account, RequestConte
 
                 try
                 {
-                    var args = new TokenCacheNotificationArgs(
-                        this,
-                        ClientId,
-                        account,
-                        true,
-                        (this as ITokenCacheInternal).IsApplicationCache,
-                        (this as ITokenCacheInternal).HasTokensNoLocks(),
-                        account.HomeAccountId.Identifier);
+                    ITokenCacheInternal tokenCacheInternal = this;
 
                     try
```

Comment:
Can you please review all these `try` blocks? They seem excessive. I can help.

Reply:
Combined the two outer try's. I think semaphore release should be in the separate outer try, in case there's exception in with OnAfterAccessAsync in the inner try.

Reply:
@pmaytak in codeflow, the issues resolved via the Github UI don't appear as resolved, can you comment on the items from Bogdan as "done" or "fixed" so it's clear they've been addressed. thanks

---
In reply to: [534843438](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2261#discussion_r534843438) [](ancestors = 534843438)

Reply:
Ah, ok. will do.
Done.